src/AsmResolver.DotNet/Cloning/MemberCloneContext.cs

```diff
@@ -12,10 +12,18 @@ public class MemberCloneContext
         /// Creates a new instance of the <see cref="MemberCloneContext"/> class.
         /// </summary>
         /// <param name="module">The target module to copy the cloned members into.</param>
-        public MemberCloneContext(ModuleDefinition module)
+        public MemberCloneContext(ModuleDefinition module) : this(module, null) { }
+
+        /// <summary>
+        /// Creates a new instance of the <see cref="MemberCloneContext"/> class.
+        /// </summary>
+        /// <param name="module">The target module to copy the cloned members into.</param>
+        /// <param name="importerInstantiator">The instantiator for creating the reference importer</param>
+        public MemberCloneContext(ModuleDefinition module,
+            Func<MemberCloneContext, CloneContextAwareReferenceImporter>? importerInstantiator)
         {
             Module = module ?? throw new ArgumentNullException(nameof(module));
-            Importer = new CloneContextAwareReferenceImporter(this);
+            Importer = importerInstantiator?.Invoke(this) ?? new CloneContextAwareReferenceImporter(this);
```

Comment:
Would a better name be a `factory` over an `instantiator`?

Reply:
I like factory better, too 👍

Reply:
Done